src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -69,11 +73,38 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
     may.add(Attr.groupSynchronizedTimestamp)
     attrs.put(may)
 
-
-    // Add the new schema object for "fooObjectClass"
+    // Add the new schema object for "workbenchGroup"
     schema.createSubcontext("ClassDefinition/workbenchGroup", attrs)
   }
 
+  def removeWorkbenchPersonSchema(): Future[Unit] = withContext { ctx =>
+    val schema = ctx.getSchema("")
+
+    // Intentionally ignores errors
+    Try { schema.destroySubcontext("ClassDefinition/workbenchPerson") }
+    Try { schema.destroySubcontext(s"AttributeDefinition/${Attr.petServiceAccount}") }
+  }
+
+  def createWorkbenchPersonSchema(): Future[Unit] = withContext { ctx =>
+    val schema = ctx.getSchema("")
+
+    createAttributeDefinition(schema, "1.3.6.1.4.1.18060.0.4.3.2.400", Attr.petServiceAccount, "pet service account of the user", true)
+
+    val attrs = new BasicAttributes(true) // Ignore case
+    attrs.put("NUMERICOID", "1.3.6.1.4.1.18060.0.4.3.2.300")
+    attrs.put("NAME", "workbenchPerson")
+    attrs.put("SUP", "inetOrgPerson")
+    attrs.put("STRUCTURAL", "true")
+
+    val must = new BasicAttribute("MUST")
+    must.add("objectclass")
+    must.add(Attr.petServiceAccount)
```

Comment:
is petServiceAccount a required attribute? Given that there is an add function seems not. This should be MAY instead of MUST

Reply:
My thinking was that it's a required attribute for the `workbenchPerson` objectclass. When a pet SA is added we add the `workbenchPerson` objectclass and when a pet SA is removed we remove the `workbenchPerson` objectclass. I can make it MAY though (in that case we would not need to remove the `workbenchPerson` objectclass right?)